packages/graphql/src/classes/Subgraph.ts

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { buildSubgraphSchema } from "@apollo/subgraph";
+import { mergeTypeDefs } from "@graphql-tools/merge";
+import { IResolvers, makeDirectiveNode, TypeSource } from "@graphql-tools/utils";
+import {
+    ConstDirectiveNode,
+    DefinitionNode,
+    DocumentNode,
+    GraphQLDirective,
+    GraphQLNamedType,
+    GraphQLResolveInfo,
+    Kind,
+    parse,
+    print,
+    SchemaExtensionNode,
+} from "graphql";
+import type * as neo4j from "neo4j-driver";
+import { translateResolveReference } from "../translate/translate-resolve-reference";
+import type { Context, Node } from "../types";
+import { execute } from "../utils";
+import getNeo4jResolveTree from "../utils/get-neo4j-resolve-tree";
+import { Executor } from "./Executor";
+
+type FederationDirectiveName =
+    | "key"
+    | "extends"
+    | "shareable"
+    | "inaccessible"
+    | "override"
+    | "external"
+    | "provides"
+    | "requires"
+    | "tag";
+
+type FullyQualifiedFederationDirectiveName = `federation__${FederationDirectiveName}`;
+
+const isFederationDirectiveName = (name: string): name is FederationDirectiveName =>
+    ["key", "extends", "shareable", "inaccessible", "override", "external", "provides", "requires", "tag"].includes(
+        name
+    );
```

Comment:
**nitpick**: Are we favouring Arrow functions for top-level functions?

Personally, I find it more readable as: 

```suggestion
function isFederationDirectiveName(name: string): name is FederationDirectiveName {
    ["key", "extends", "shareable", "inaccessible", "override", "external", "provides", "requires", "tag"].includes(
        name
    );
}
```

As an extra nitpick, consider moving the array outside the function to avoid re-creating them on every call, which happens on every loop. I don't think it has a significant impact on performance, so not an important change

On a different note, `name is FederationDirectiveName` is *chef-kiss*

Reply:
For superior performance, if this ends up at some point being noticeable, a `Set` is the appropriate structure for this

Reply:
Honestly, for a single statement function like this, I quite prefer an arrow function. And strangely, I especially find the `x is y` return type to be more readable in an arrow function, but it's really personal preference! I can change it.

Array is now defined outside and the type is inferred from the array.